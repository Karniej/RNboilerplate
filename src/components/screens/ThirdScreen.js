// @flow

import React from 'react';
import { StyleSheet, Text, View, Button } from 'react-native';

class ThirdScreen extends React.Component<{ navigation: any }> {
  static navigationOptions = {
    title: 'Third',
  };
  render() {
    const { navigate } = this.props.navigation;
    return (
      <View style={styles.container}>
        <Text style={styles.welcome}>Welcome to Third Screen</Text>
        <Button title="Second" onPress={() => navigate('Second')} />
      </View>
    );
  }
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center',
    backgroundColor: 'violet',
  },
  welcome: {
    fontSize: 20,
    textAlign: 'center',
    margin: 10,
  },
});

export default ThirdScreen;
