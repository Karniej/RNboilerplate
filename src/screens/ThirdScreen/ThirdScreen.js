import React from 'react'
import PropTypes from 'prop-types'

import { ContentWrapper, Header, MainButton } from '../../components'
import { colors } from '../../config'

class ThirdScreen extends React.Component {
  static navigationOptions = {
    title: 'Third',
    drawerLabel: 'Third',
    headerStyle: {
      backgroundColor: colors.secondaryColor
    }
  }
  render() {
    const { navigation } = this.props

    return (
      <ContentWrapper>
        <Header
          title='Third Screen'
        />
        <MainButton
          title='Log Out'
          navigation={navigation}
          resetStack
          routeName='LoginStack'
        />
      </ContentWrapper>
    )
  }
}
ThirdScreen.propTypes = {
  navigation: PropTypes.shape({
    dispatch: PropTypes.func
  }).isRequired
}
export default ThirdScreen
