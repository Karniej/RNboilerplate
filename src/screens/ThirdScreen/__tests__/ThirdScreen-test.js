import renderer from 'react-test-renderer';
import React from 'react';
import ThirdScreen from '../ThirdScreen';

test('renders correctly', () => {
  const navigation = { navigate: jest.fn() };
  const onPress = jest.fn();
  expect(onPress).toBeDefined();
  expect(onPress).not.toBeCalled();
  const tree = renderer.create(<ThirdScreen navigation={navigation} onPress={onPress} />).toJSON();
  expect(tree).toMatchSnapshot();
});
