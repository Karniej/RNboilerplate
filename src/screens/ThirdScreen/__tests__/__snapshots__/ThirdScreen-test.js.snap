// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#F5FCFF",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "fontSize": 20,
        "margin": 10,
        "textAlign": "center",
      }
    }
  >
    Welcome to Third Screen
  </Text>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    collapsable={undefined}
    hasTVPreferredFocus={undefined}
    hitSlop={undefined}
    isTVSelectable={true}
    nativeID={undefined}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
    testID="logout"
    tvParallaxProperties={undefined}
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "#00d664",
          "fontSize": 18,
        }
      }
    >
      Logout
    </Text>
  </View>
</View>
`;
